### PR TITLE
chore(field-permissions): Implement FieldPermissions (allow to nullify field if you have no permissions)

### PR DESCRIPTION
**Pull request by tomaszdurka**
…y field if you have no permissions)

### CODE REVIEW

File: libs/member-plan/api/src/lib/member-plan.model.ts [hunk 1]
```diff
@@ -32,7 +33,7 @@ export class MemberPlan {
   @Field()
   description!: string
 
-  @Field()
+  @FieldProtected(() => Boolean, {permissions: [CanGetMemberPlan]})
```
Comment (Itrulia) [inline libs/member-plan/api/src/lib/member-plan.model.ts:36]: Can probably always be returned and only query active memberplans for users with the perms instead

File: libs/navigation/api/src/lib/navigation.model.ts [hunk 2]
```diff
@@ -33,7 +34,7 @@ export class Navigation {
   @Field()
   key!: string
 
-  @Field(() => [BaseNavigationLink])
+  @FieldProtected(() => [BaseNavigationLink], {permissions: [CanGetNavigations]})
```
Comment (Itrulia) [inline libs/navigation/api/src/lib/navigation.model.ts:37]: Should probably always be returned :D

File: libs/member-plan/api/src/lib/member-plan.model.ts (hunk 1 above)
Comment (Itrulia) [reply libs/member-plan/api/src/lib/member-plan.model.ts:36]: I wonder if we can introspect the type for boolean, number etc, so we don't always have to specify.

File: libs/permissions/api/src/lib/field-permissions.middleware.ts [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+export const FieldMiddlewarePermissions: FieldMiddleware = async (ctx, next) => {
+  const {info} = ctx
+  const {extensions} = info.parentType.getFields()[info.fieldName]
+  const permissions = extensions.permissions as Permission[] | undefined
+  const session = ctx.context.req.user
+
+  if (!permissions) {
+    return next()
+  }
```
Comment (Itrulia) [inline libs/permissions/api/src/lib/field-permissions.middleware.ts:13]:
```suggestion
  }
  
```

File: libs/navigation/api/src/lib/navigation.model.ts (hunk 2 above)
Comment (tomaszdurka) [reply libs/navigation/api/src/lib/navigation.model.ts:37]: Yes indeed, just added this to illustrate.

File: libs/member-plan/api/src/lib/member-plan.model.ts (hunk 1 above)
Comment (tomaszdurka) [reply libs/member-plan/api/src/lib/member-plan.model.ts:36]: Agree, it's not really harmful to have it.

**Pull request merged**